### DISCUSSION TITLE
Payment Providers

### DISCUSSION THREAD

**Discussion by itscark** [shopware/frontends#568] [Ideas]
The seamless integration of multiple payment providers in Shopware frontends is crucial for an optimized checkout process. Creating a <CheckoutPaymentForm /> component that dynamically displays payment options for different providers could be the solution.

There is currently no universal solution in Shopware frontends for the simple integration of different payment providers. By offering a customizable component such as <CheckoutPaymentForm />, the currently existing examples for Mollie or Adyen could be adapted into a simple entry-level solution.

We have developed a working example for the integration of a payment provider. If there is interest, I am happy to customize our solution for the Vue demo store and submit a pull request for review. Perhaps it will help others and foster an environment of shared progress and innovation in payment integrations.

**Comment by mkucmus** [comment]
hey @itscark, that sounds very promising!
You are right, currently there is no general solution and we didn't even try to be agnostic in terms of payments as we realized that  payment providers may vary a lot, thus it will make the vue-demo-store template much harder to read and understand - that's why we provide standalone examples apart, but as you have notice, it does not cover every case (let alone all-in-one entrypoint).  

However! we are very curious how you handled that and looking forward for some PR 💙 I believe it will speed up the building process for everyone and will probably make you guys a Composable Frontends Superstars 🌟

**Comment by BrocksiNet** [comment]
Yeah great to hear 💪 We can also add your Payment Integration (Example) to our new nice [Doc-Page](https://shopware-frontends-docs-git-feat-docs-5a6821-shopware-frontends.vercel.app/resources/integrations/payments/) (that is currently under development). 🤗
